packages/astro/src/core/errors/errors-data.ts

```diff
@@ -1563,6 +1563,34 @@ export const InvalidContentEntryDataError = {
 	hint: 'See https://docs.astro.build/en/guides/content-collections/ for more information on content schemas.',
 } satisfies ErrorData;
 
+/**
+ * @docs
+ * @message
+ * **Example error message:**<br/>
+ * The content loader for the collection **blog** returned an invalid result:<br/>
+ * Content entry `id` must be a string:<br/>
+ * {<br/>
+ * 	 "id": 1,<br/>
+ * 	 "title": "Hello, World!"<br/>
+ * }
+ * @description
+ * A content loader returned an invalid result.
+ * Make sure that the ID of the entry is a string.
+ * See the [Content collections documentation](https://docs.astro.build/en/guides/content-collections/) for more information.
+ */
+export const ContentLoaderReturnsInvalidResult = {
```

Comment:
```suggestion
export const ContentLoaderReturnsInvalidId = {
```
Just an e.g. based on my earlier question. There are probably lots of ways an entry can be invalid, so does it make sense to entirely base this on invalid ID

Reply:
See reply on previous change request